tsconfig.json

```diff
@@ -0,0 +1,56 @@
+{
+  "compilerOptions": {
+    "target": "es2017",
+    "allowJs": true,
```

Comment:
Why do we need `allowJs`? I've had issues with TS errors being ignored until publish time (and breaking publishing automation) because `allowJs` is explicitly set to `false` by `ts:precompile` here

https://github.com/typed-ember/ember-cli-typescript/blob/v3.1.3/ts/lib/commands/precompile.ts#L31

Reply:
This is a good catch. I will have to dig into the history to see why we set that default, and likely change it – upcoming 4.0 release would be a good time.

Reply:
Fixed!